File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* temp=head;
   

        if(!head or !head->next){
            return NULL;
        }
        int nodes=0;
        
        while(temp!=NULL){
            nodes++;
            temp=temp->next;
    }

        int c=1;
        int pos=nodes-n;

        if(n==nodes){
            head=head->next;
        }
    else
    {temp=head;
    while(c!=pos and temp!=NULL ){
        temp=temp->next;
        c++;
    }

    temp->next=temp->next->next;}

    return head;
        
        
    }
};

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null || head.next==null) return null;    
        int c=0;                                          
        ListNode temp=head;
        while(temp!=null){
            temp=temp.next;
            c++;
        }
        if(n==c) return head.next;                        
        temp=head;
        for(int i=0; i<c-n-1; i++) temp=temp.next;
        if(n==1) temp.next=null;                          
        else temp.next=temp.next.next;                    
        return head;}}